src/components/text-editor/text-editor.pw.tsx

```diff
@@ -789,3 +836,531 @@ test("should set aria-live attribute on Character Count to `polite` when compone
+    await textEditorInput(page).pressSequentially("bar");
+    await page.keyboard.press("Enter");
+    await textEditorInput(page).pressSequentially("baz");
+
+    await expect(innerTextList(page, "ol", 1)).toHaveCount(1);
+    await expect(innerTextList(page, "ol", 1)).toHaveText("foo");
+    await expect(innerTextList(page, "ol", 2)).toHaveCount(1);
+    await expect(innerTextList(page, "ol", 2)).toHaveText("bar");
+    await expect(innerTextList(page, "ol", 3)).toHaveCount(1);
+    await expect(innerTextList(page, "ol", 3)).toHaveText("baz");
+  });
+
+  test("when the `characterLimit` prop is specified, key presses have no effect when the length of entered text has reached the limit", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TextEditorCustom characterLimit={10} />);
+
+    await textEditorInput(page).fill("1234567890");
+    await expect(innerText(page)).toHaveText("1234567890");
+    await textEditorInput(page).pressSequentially("1");
+    await expect(innerText(page)).toHaveText("1234567890");
+  });
+
+  test("content can be pasted into the editor if it would not exceed the character limit", async ({
+    mount,
+    page,
+    context,
+  }) => {
+    // grant access to clipboard
+    await context.grantPermissions(["clipboard-read", "clipboard-write"]);
+    await mount(<TextEditorCustom characterLimit={10} />);
+    await textEditorInput(page).focus();
+    // copy text to clipboard
+    await page.evaluate(() => navigator.clipboard.writeText("1234567890"));
+
+    if (process.env.CI) {
+      // assume Ubuntu (which is the OS the Github actions run on), where Control+Shift+V is needed to paste
+      await page.keyboard.down("Control");
+      await page.keyboard.down("Shift");
+      await page.keyboard.press("v");
+      await page.keyboard.up("Shift");
+      await page.keyboard.up("Control");
+    } else {
+      // assuming MacOS, press Meta(Command)+V to paste
+      await page.keyboard.down("Meta");
+      await page.keyboard.press("v");
+      await page.keyboard.up("Meta");
+    }
+
+    await expect(innerText(page)).toHaveText("1234567890");
+  });
+
+  test("content pasted into the editor whose length exceed the character limit is cut short", async ({
+    mount,
+    page,
+    context,
+  }) => {
+    // grant access to clipboard
+    await context.grantPermissions(["clipboard-read", "clipboard-write"]);
+    await mount(<TextEditorCustom characterLimit={10} />);
+    await textEditorInput(page).focus();
+    // copy text to clipboard
+    await page.evaluate(() =>
+      navigator.clipboard.writeText("12345678901234567890")
+    );
+
+    if (process.env.CI) {
+      // assume Ubuntu (which is the OS the Github actions run on), where Control+Shift+V is needed to paste
+      await page.keyboard.down("Control");
+      await page.keyboard.down("Shift");
+      await page.keyboard.press("v");
+      await page.keyboard.up("Shift");
+      await page.keyboard.up("Control");
```

Comment:
Same as previous comment, could you add the following for anyone who might be running tests locally on Windows?
```
} else if (process.platform === "win32") {
  // Windows: Control+V
  await page.keyboard.down("Control");
  await page.keyboard.press("v");
  await page.keyboard.up("Control");
```